packages/js/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -33,6 +34,19 @@ export default async function runExecutor(
   context: ExecutorContext
 ) {
   const pm = detectPackageManager();
+
+  let isNpmInstalled = false;
+  try {
+    // If npm is installed, we should get a version string, otherwise an error will be thrown.
+    isNpmInstalled =
+      getPackageManagerVersion('npm', process.cwd(), true) !== '';
+  } catch (e) {
+    // Allow missing npm only when using bun (https://github.com/nrwl/nx/pull/32252#discussion_r2262481795)
+    if (pm !== 'bun') {
+      throw e;
```

Comment:
I'd rather have this be a controlled user-facing message and handled error state please:

```ts
console.error(`npm was not found in the current environment. This is only supported when using \`bun\` as a package manager, but your detected package manager is "${pm}"`);
return {
  success: false,
};
```